overview/content-types/option.nix

```diff
@@ -0,0 +1,123 @@
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}:
+let
+  inherit (lib)
+    types
+    mkOption
+    join
+    drop
+    optionalString
+    take
+    ;
+in
+{
+  options = {
+    prefix-length = mkOption {
+      type = types.int;
+      default = 2;
+    };
+    loc = mkOption {
+      type = with types; listOf str;
+    };
+    type = mkOption {
+      type = types.str;
+    };
+    default = mkOption {
+      type = types.attrs;
+      default = { };
+    };
+    description = mkOption {
+      type = types.str;
+    };
+    readOnly = mkOption {
+      type = types.bool;
+    };
+    option-prefix = mkOption {
+      type = types.str;
+      readOnly = true;
+      default =
+        let
+          prefix-head = take config.prefix-length config.loc;
+          prefix-tail = drop config.prefix-length config.loc;
+        in
+        ''
+          <span class="option-prefix">${join "." prefix-head}.</span><span>${join "." prefix-tail}</span>
+        '';
+    };
+    option-type = mkOption {
+      type = types.str;
+      readOnly = true;
+      default = ''
+        <dt>Type:</dt>
+        <dd class="option-type"><code>${config.type}</code></dd>
+      '';
+    };
+    option-default = mkOption {
+      type = types.str;
+      readOnly = true;
+      default = optionalString (config.default ? text) ''
+        <dt>Default:</dt>
+        <dd class="option-default"><code>${config.default.text}</code></dd>
+      '';
+    };
+    option-description = mkOption {
+      type = types.str;
+      readOnly = true;
+      default =
+        let
+          # This doesn't actually produce a HTML string but a Jinja2 template string
+          # literal, that is then replaced by it's HTML translation at the last build
+          # step.
```

Comment:
Add the rationale: this avoids IFD since this way we can do markdown processing in a single step at the end